util/fetch/src/client.rs

```diff
@@ -199,16 +231,34 @@ impl Client {
 		})
 	}
 
-	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>) -> io::Result<thread::JoinHandle<()>> {
+	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>, https_proxy_url: String) -> io::Result<thread::JoinHandle<()>> {
 		thread::Builder::new().name("fetch".into()).spawn(move || {
 			let mut core = match reactor::Core::new() {
 				Ok(c) => c,
 				Err(e) => return tx_start.send(Err(e)).unwrap_or(())
 			};
 
+			let connector = {
+				let https_connector = hyper_rustls::HttpsConnector::new(4, &core.handle());
+
+				match https_proxy_url.parse() {
```

Comment:
use map().parse() as I suggested earlier i.e, if the String is `None` then directly indicate an error instead of creating an empty String (default)